### PR TITLE
REST: Include suppressed exceptions on failures

### LINKED ISSUE #23392: Include suppressed exceptions in rest response?
On the transport layer suppressed exceptions are being kept around (in `ElasticsearchException#readStackTrace(...)` and `Elasticsearch#writeStackTraces(...)`), so maybe we should include suppressed exceptions in rest response too? Or at least give the ability to do this via a rest parameter.

In the put pipeline api this can be useful as all the config errors can then be returned instead of failing on the first configuration error. Obviously other apis can benefit from this as well.

### PR DESCRIPTION
**Pull request by nik9000**
This modifies xcontent serialization of Exceptions to contain suppressed
exceptions. If there are any suppressed exceptions they are included in
the exception response by default. The reasoning here is that they are
fairly rare but when they exist they almost always add extra useful
information. Take, for example, the response when you specify two broken
ingest pipelines:

```
{
  "error" : {
    "root_cause" : ...snip...
    "type" : "parse_exception",
    "reason" : "[field] required property is missing",
    "header" : {
      "processor_type" : "set",
      "property_name" : "field"
    },
    "suppressed" : [
      {
        "type" : "parse_exception",
        "reason" : "[field] required property is missing",
        "header" : {
          "processor_type" : "convert",
          "property_name" : "field"
        }
      }
    ]
  },
  "status" : 400
}
```

Moreover, when suppressed exceptions come from 500 level errors should
give us more useful debugging information.

Closes #23392

### CODE REVIEW

**Review by javanna — APPROVED** [review]
nice thanks @nik9000 LGTM

**Review by jasontedor — APPROVED** [review]
Great change.

File: server/src/main/java/org/elasticsearch/ElasticsearchException.java [hunk 1]
```diff
@@ -384,6 +374,17 @@ protected static void innerToXContent(XContentBuilder builder, Params params,
...
+        Throwable[] allSuppressed = throwable.getSuppressed();
+        if (allSuppressed.length > 0) {
+            builder.startArray(SUPPRESSED.getPreferredName());
+            for (Throwable suppressed : allSuppressed) {
+                builder.startObject();
+                generateThrowableXContent(builder, params, suppressed);
+                builder.endObject();
+            }
+            builder.endArray();
+        }
```
Comment (javanna) [inline server/src/main/java/org/elasticsearch/ElasticsearchException.java]: out of curiosity, does order matter here? Why moving suppressed?

Comment (nik9000) [reply server/src/main/java/org/elasticsearch/ElasticsearchException.java]: I think it is easier to read when the suppressed comes after the headings. That keeps everything about each exception close together. To be honest, if it is important to me, I should add a unit test for it. I'll do that.

**Review by martijnvg — APPROVED** [review]
:+1:  @nik9000 Thanks for adding this!

**Comment by nik9000** [comment]
> +1 @nik9000 Thanks for adding this!

Thanks for filing the issue!

**Pull request merged**